File: leetcode3/황은지/403. Frog Jump.js
/**
 * @param {number[]} stones
 * @return {boolean}
 */
var canCross = function(stones) {
    const dp =Array.from({length:stones.length},()=>Array(3).fill(false));
    dp[1][1]=true;

// 길이는 전길이에 따라 달라짐
    for(let i=0;i<stones.length-1;i++){
        const pos=stones[i];
        for(let j=0;j<3;j++){
            if(dp[i][j]===true){
            } 
        }
    }

    return  true;
};

File: leetcode3/최원준/403. Frog Jump.py
class Solution:
    def canCross(self, stones: List[int]) -> bool:
        stone_set = set(stones)
        target = stones[-1]

        visited = set()

        def dfs(pos, k):
            if pos == target:
                return True

            if (pos, k) in visited:
                return False

            visited.add((pos, k))

            for next_k in [k - 1, k, k + 1]:
                if next_k <= 0:
                    continue

                next_pos = pos + next_k

                if next_pos in stone_set:
                    if dfs(next_pos, next_k):
                        return True

            return False

        return dfs(0, 0)
